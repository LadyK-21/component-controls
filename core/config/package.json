{
  "name": "@component-controls/config",
  "version": "3.11.0",
  "description": "Component controls configuration reading utilities",
  "main": "dist/index.js",
  "module": "dist/index.esm.js",
  "typings": "dist/index.d.ts",
  "files": [
    "dist/",
    "package.json",
    "README.md"
  ],
  "scripts": {
    "build": "yarn cross-env NODE_ENV=production rollup -c",
    "dev": "yarn cross-env NODE_ENV=development rollup -cw",
    "docs": "ts-md",
    "fix": "yarn lint --fix",
    "lint": "yarn eslint . --ext mdx,ts,tsx",
    "prepare": "yarn build",
    "test": "yarn jest"
  },
  "homepage": "https://github.com/ccontrols/component-controls",
  "bugs": {
    "url": "https://github.com/ccontrols/component-controls/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/ccontrols/component-controls.git",
    "directory": "core/config"
  },
  "license": "MIT",
  "dependencies": {
    "@component-controls/core": "^3.11.0",
    "@component-controls/instrument": "^3.11.0",
    "@component-controls/render": "^3.11.0",
    "glob": "^7.1.6",
    "glob-base": "^0.3.0",
    "micromatch": "^4.0.2",
    "typescript": "^4.0.5",
    "yargs": "^15.3.1"
  },
  "devDependencies": {
    "@component-controls/ts-markdown-docs": "^3.8.2",
    "@types/faker": "^4.1.9",
    "@types/glob": "^7.1.3",
    "@types/glob-base": "^0.3.0",
    "@types/jest": "^26.0.10",
    "@types/micromatch": "^4.0.1",
    "cross-env": "^5.2.1",
    "eslint": "^6.5.1",
    "jest": "^26.4.2"
  },
  "esm": {
    "cjs": true
  },
  "browser": {
    "fs": false
  },
  "jest": {
    "testEnvironment": "node",
    "roots": [
      "."
    ]
  },
  "publishConfig": {
    "access": "public"
  },
  "authors": [
    "Atanas Stoyanov"
  ],
  "gitHead": "e30d62f101e104711a16261fce04785d58cac1eb"
}
