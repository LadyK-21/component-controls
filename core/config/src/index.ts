import * as path from 'path';
import * as fs from 'fs';

import { sync as globSync } from 'glob';
import globBase from 'glob-base';
import { makeRe } from 'micromatch';

import yargs from 'yargs';
import { BuildConfiguration } from '@component-controls/core';

export const buildConfigFileNames = [
  'buildtime.js',
  'build.js',
  'main.js',
  'doczrc.js',
];

export const optionsFileNames = [
  'runtime.js',
  'options.js',
  'runtime.ts',
  'runtime.tsx',
  'options.ts',
  'options.tsx',
];
export interface ConfigurationResult {
  config: BuildConfiguration;
  configPath: string;
  optionsFilePath?: string;
}
/**
 * return the configuration folder from command-line parameters
 * command line accepts -c/ -config parameter for config path
 * the config file is assumed named main.js/main.ts
 * @param baseFolder project folder to start the search with
 */
export const getConfigurationArg = (
  args: string[] = process.argv,
): string | undefined => {
  const argv = yargs(args)
    .option('config', {
      alias: 'c',
      description: 'configuration folder',
      type: 'string',
    })
    .help()
    .alias('help', 'h').argv;
  return argv.config;
};

/**
 *  given a base project folder and a configuration folder, returns the configuration file
 *
 * @param baseFolder project folder to start the search with
 * @param configFolder folder where the configuration file is located
 * @param args optional arguments
 */
export const loadConfiguration = (
  baseFolder: string,
  configFolder?: string,
  args?: string[],
  defaultConfigPath?: string,
): ConfigurationResult | undefined => {
  const folder = configFolder ?? getConfigurationArg(args) ?? defaultConfigPath;
  const configPath = folder ? path.resolve(baseFolder, folder) : baseFolder;
  const hasConfigFolder = fs.existsSync(configPath);
  if (!hasConfigFolder) {
    console.warn('configuration folder not found', configPath);
  }
  const allFiles = hasConfigFolder ? fs.readdirSync(configPath) : [];
  const buildConfigFile = allFiles.find(file =>
    buildConfigFileNames.includes(file.toLowerCase()),
  );
  const optionsFile = allFiles.find(file =>
    optionsFileNames.includes(file.toLowerCase()),
  );
  if (buildConfigFile) {
    const buildPath = path.resolve(configPath, buildConfigFile);
    let config = require('esm')(module)(buildPath);
    if (
      !config ||
      (typeof config === 'object' && Object.keys(config).length === 0)
    ) {
      config = require(buildPath);
    }
    return {
      config: config.default || config,
      optionsFilePath: optionsFile
        ? path.resolve(configPath, optionsFile)
        : undefined,
      configPath,
    };
  }
  return undefined;
};

//fix for sb5 issue handling glob
const fixGlob = (golbExpr: string): string => golbExpr.replace('.(', '.@(');

/**
 * find the story files out of a configuration file
 * using glob for the regex file search
 */
export const extractDocuments = ({
  config,
  configPath,
}: ConfigurationResult): string[] | undefined => {
  const stories = config ? config.stories || config.files : undefined;
  const files = typeof stories === 'string' ? [stories] : stories;
  const documents = files
    ? files.reduce((acc: string[], storyRg: string) => {
        const matches = globSync(path.resolve(configPath, fixGlob(storyRg)));
        if (!matches.length) {
          throw new Error(`${storyRg} did not match any files`);
        }
        const ignore = config.ignore || [];
        const files = matches.filter(filename => {
          const basename = path.basename(filename).toLowerCase();
          return !ignore.includes(basename);
        });
        return [...acc, ...files];
      }, [])
    : undefined;
  return documents;
};

/**
 * from the glob list of documents, extract require.context array of props
 */
export interface RequireContextProps {
  directory: string;
  useSubdirectories: boolean;
  regExp: RegExp;
}
/**
 * convert glob patters from config file into require.context objects
 */
export const configRequireContext = ({
  config,
  configPath,
}: ConfigurationResult): RequireContextProps[] | undefined => {
  const stories = config ? config.stories || config.files : undefined;
  const files = typeof stories === 'string' ? [stories] : stories;

  const contexts = files
    ? files.reduce((acc: RequireContextProps[], storyRg: string) => {
        const base = globBase(fixGlob(storyRg));
        const useSubdirectories = base.glob.startsWith('**');
        const glob = useSubdirectories ? base.glob.substr(3) : base.glob;
        const regExp = base.isGlob
          ? RegExp(makeRe(glob).source.substr(1))
          : new RegExp(glob);
        return [
          ...acc,
          {
            directory: path.resolve(configPath, base.base),
            useSubdirectories,
            regExp,
          },
        ];
      }, [])
    : undefined;
  return contexts;
};

/**
 * merge a configuration passed through cli or tools, with the build configration from the config path
 */
export const mergeBuildConfiguration = (
  config: BuildConfiguration,
): BuildConfiguration => {
  const { configPath } = config;
  const buildConfig = loadConfiguration(process.cwd(), configPath);
  return { ...buildConfig?.config, ...config };
};
