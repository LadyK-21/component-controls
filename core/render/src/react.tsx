/* eslint-disable react/display-name */
import { createElement } from 'react';
import {
  StoryRenderFn,
  getControlValues,
  deepMerge,
  FrameworkRenderFn,
} from '@component-controls/core';

export const render: FrameworkRenderFn = (story, doc, options: any = {}) => {
  if (!story) {
    throw new Error(`Invalid story`);
  }
  if (!doc) {
    throw new Error(`Invalid doc`);
  }
  const controls = story.controls;
  const values = getControlValues(controls);
  //parameters added to avoid bug in SB6 rc that assumes parameters exist
  const context = {
    story,
    doc,
    controls,
    ...options,
  };
  const { decorators: globalDecorators = [] } = options;
  const { decorators: storyDecorators = [] } = story;
  const decorators = deepMerge<StoryRenderFn[]>(
    globalDecorators,
    storyDecorators,
  );
  const sortedDecorators = decorators.reverse();
  let renderFn = story.renderFn;
  for (let i = 0; i < sortedDecorators.length; i += 1) {
    const decorator = sortedDecorators[i];
    const childFn = renderFn;
    const nextRenderFn = (_: any, nextContext: any) =>
      (childFn as StoryRenderFn)(values, { ...context, ...nextContext });
    renderFn = () =>
      decorator(nextRenderFn, {
        ...context,
        renderFn: nextRenderFn,
      });
  }
  return typeof renderFn === 'function'
    ? createElement(() => (renderFn as StoryRenderFn)(values, context))
    : createElement('div', 'invalid render function');
};
