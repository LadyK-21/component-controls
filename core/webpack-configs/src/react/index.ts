import path from 'path';
import { RuleSetLoader } from 'webpack';
import ExtractCssPlugin from 'extract-css-chunks-webpack-plugin';
import OptimizeCssAssetsWebpackPlugin from 'optimize-css-assets-webpack-plugin';

import {
  PresetType,
  BuildProps,
  defCssFileName,
  customLoaderOptions,
} from '@component-controls/core';
import { findUpFile } from '@component-controls/core/node-utils';

export const react: PresetType = (options: BuildProps) => {
  const isProd = process.env.NODE_ENV === 'production';
  const cssLoaders: RuleSetLoader[] = [];
  const postcssOptions = customLoaderOptions(options, 'postcss-loader', {});
  const postCssOptionsFile = findUpFile(process.cwd(), 'postcss.config.js');
  const hasPostCss = Object.keys(postcssOptions).length || postCssOptionsFile;
  if (hasPostCss && !((postcssOptions as any).disable === true)) {
    cssLoaders.push({
      loader: 'postcss-loader',
      options: {
        postcssOptions:
          typeof postCssOptionsFile === 'string'
            ? require(postCssOptionsFile)
            : undefined,
        sourceMap: true,
      },
    });
  }
  const result: PresetType = {
    plugins: [
      new ExtractCssPlugin({
        filename: options.cssFileName || defCssFileName,
      }),
    ],
    optimization: { minimizer: [] },
    performance: { hints: false },
    module: {
      rules: [
        {
          test: /\.(js|jsx)$/,
          exclude: [/node_modules/],
          use: [
            {
              loader: 'babel-loader',
              options: {
                presets: [
                  [
                    '@babel/preset-env',
                    {
                      targets: {
                        browsers: ['last 5 versions', 'ie >= 9'],
                        node: 'current',
                      },
                      modules: 'commonjs',
                      useBuiltIns: 'usage',
                      corejs: 3,
                    },
                  ],
                  ['@babel/preset-react', { runtime: 'automatic' }],
                ],
              },
            },
          ],
        },
        {
          test: /\.(ts|tsx)$/,
          exclude: [/node_modules/],
          use: [
            {
              loader: 'babel-loader',
              options: {
                presets: [['react-app', { flow: false, typescript: true }]],
              },
            },
          ],
        },
        {
          test: /\.(md|mdx)$/i,
          exclude: [/node_modules/],
          loader: 'babel-loader',
          options: {
            presets: [
              [
                '@babel/preset-env',
                {
                  targets: {
                    browsers: ['last 5 versions', 'ie >= 9'],
                    node: 'current',
                  },
                  modules: 'commonjs',
                  useBuiltIns: 'usage',
                  corejs: 3,
                },
              ],
              '@babel/preset-react',
            ],
          },
        },
        {
          test: /\.(eot|md|svg|ico|jpg|jpeg|png|gif|ttf|woff|woff2|pdf|mp4|web|wav|mp3|m4a|aac|oga)$/i,
          exclude: [/node_modules/],
          loader: 'url-loader',
          options: customLoaderOptions(options, 'url-loader', {
            limit: 25000,
            name: '[name].[ext]',
            publicPath: '/static',
            outputPath: path.relative(
              options?.distFolder || process.cwd(),
              path.resolve(options?.staticFolder || process.cwd()),
            ),
          }),
        },
        {
          test: /\.(woff(2)?|ttf|eot|svg)(\?v=\d+\.\d+\.\d+)?$/,
          use: [
            {
              loader: 'file-loader',
              options: customLoaderOptions(options, 'file-loader', {
                name: '[name].[ext]',
                outputPath: path.relative(
                  options?.distFolder || process.cwd(),
                  path.resolve(options?.staticFolder || process.cwd()),
                ),
              }),
            },
          ],
        },
        {
          test: /\.(css|sass|scss|less)$/i,
          use: [
            // Creates `style` nodes from JS strings
            // will export to a consolidated css file
            ExtractCssPlugin.loader,
            {
              // Translates CSS into CommonJS
              loader: 'css-loader',
              options: customLoaderOptions(options, 'css-loader', {
                sourceMap: true,
              }),
            },
            ...cssLoaders,
            {
              // Compiles Sass to CSS
              loader: 'sass-loader',
              options: customLoaderOptions(options, 'sass-loader', {
                sourceMap: true,
              }),
            },
            {
              // Compiles less to CSS
              loader: 'less-loader',
              options: customLoaderOptions(options, 'less-loader', {
                sourceMap: true,
              }),
            },
          ],
        },
        {
          test: /\.txt$/i,
          use: [
            {
              loader: 'raw-loader',
              options: customLoaderOptions(options, 'raw-loader', {}),
            },
          ],
        },

        {
          test: /\.(md|mdx)$/i,
          exclude: [/node_modules/],
          loader: '@component-controls/loader/loader',
          enforce: 'pre',
          options: {
            mdx: {
              transformMDX: true,
            },
          },
        },
        {
          test: /\.(story|stories|doc|docs).(js|jsx|ts|tsx)$/,
          loader: '@component-controls/loader/loader',
          exclude: [/node_modules/],
          enforce: 'pre',
        },
      ],
    },
    resolve: {
      extensions: ['.ts', '.tsx', '.js', '.jsx'],
    },
  };
  if (isProd) {
    result.optimization = {
      minimizer: [
        new OptimizeCssAssetsWebpackPlugin({
          cssProcessorOptions: {
            discardComments: { removeAll: true },
          },
        }),
      ],
    };
  }
  return result;
};
