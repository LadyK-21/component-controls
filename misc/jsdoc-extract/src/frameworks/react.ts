import { FrameworkPlugin, JSDocType, JSDocTypes, JSImports } from '../utils';
import { walkProps } from '../utils/resolve-props';

const reactImport = (
  names: JSDocType[],
  imports: JSImports,
): JSDocType | undefined => {
  const reactImports = imports.filter(i => i.module === 'react');
  const foundReact = names?.find(({ name }) => {
    const nameParts = name?.split('.');
    if (nameParts) {
      const namedImport = nameParts.pop();
      const defaultImport = nameParts.length ? nameParts[0] : null;
      if (namedImport) {
        return reactImports.some(
          i => i.name === defaultImport || i.namedImports?.[namedImport],
        );
      }
    }
    return false;
  });
  return foundReact;
};
const reactClassComponent = (
  component: JSDocType,
  imports: JSImports,
): JSDocType | undefined => {
  const classes = component.inherits?.filter(i => i.type === 'class' && i.name);
  if (classes) {
    return reactImport(classes, imports);
  }
  return undefined;
};

const reactFunctionComponent = (
  component: JSDocType,
  imports: JSImports,
): JSDocType | undefined => {
  if (component.type === 'function') {
    if (component.value) {
      return reactImport([component.value], imports);
    }
    if (component.parameters?.length) {
      return {
        parameters: component.parameters,
      };
    }
  }

  return undefined;
};
const reactComponent = (
  component: JSDocType,
  imports: JSImports,
): JSDocType | undefined => {
  return (
    reactClassComponent(component, imports) ||
    reactFunctionComponent(component, imports)
  );
};
const componentToProps = (
  jsDocs: JSDocTypes,
  component: JSDocType,
  typeProp: JSDocType,
): JSDocType[] => {
  if (typeProp.parameters?.length) {
    return walkProps(jsDocs, typeProp.parameters[0]);
  }
  return [];
};
export const extractProps: FrameworkPlugin = parse => {
  return Object.keys(parse.structures).reduce((acc, name) => {
    const component = parse.structures[name];
    const typeProp = reactComponent(component, parse.imports);
    if (typeProp) {
      const reactComponent = { ...component };
      reactComponent.properties = componentToProps(
        parse.structures,
        component,
        typeProp,
      );
      delete reactComponent.parameters;
      delete reactComponent.inherits;
      delete reactComponent.value;
      return { ...acc, [name]: reactComponent };
    }
    return { ...acc, [name]: component };
  }, {});
};
