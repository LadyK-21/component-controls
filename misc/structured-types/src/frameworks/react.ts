import * as ts from 'typescript';
import { getSymbolType, getFunctionLike, getInitializer } from '../ts-utils';
import {
  isObjectTypeDeclaration,
  getObjectStaticProp,
  ParsePlugin,
} from '../ts-utils';
import { PropKind } from '../types';

const reactPLugin: ParsePlugin = {
  internalTypes: ['PropsWithChildren'],
  typesResolve: ({ symbolType, declaration, checker }) => {
    let propsType = symbolType;
    let initializer: ts.Node | undefined = undefined;
    let name = undefined;
    let kind = undefined;
    let framework = undefined;
    let defaultProps = undefined;
    let displayName = undefined;
    if ((symbolType.flags & ts.TypeFlags.Object) === ts.TypeFlags.Object) {
      if (declaration) {
        if (isObjectTypeDeclaration(declaration)) {
          initializer = getInitializer(declaration);
          const jsx = checker.getJsxIntrinsicTagNamesAt(declaration);
          if (jsx) {
            defaultProps = getObjectStaticProp(declaration, 'defaultProps');
            displayName = getObjectStaticProp(declaration, 'displayName');

            framework = 'react';
            kind = PropKind.Class;
            const signatures = symbolType.getConstructSignatures();
            if (signatures.length > 0 && signatures[0].parameters.length) {
              const props = signatures[0].parameters[0];
              propsType = getSymbolType(checker, props) || symbolType;

              if (propsType.isUnionOrIntersection()) {
                propsType = propsType.types[0];
              }
            }
          }
        } else {
          const reactFunction = getFunctionLike(checker, declaration);
          if (reactFunction) {
            const jsx = checker.getJsxIntrinsicTagNamesAt(reactFunction);
            if (jsx) {
              defaultProps = getObjectStaticProp(
                reactFunction.parent,
                'defaultProps',
              );
              displayName =
                getObjectStaticProp(reactFunction.parent, 'displayName') ||
                reactFunction.name?.getText() ||
                (ts.isVariableDeclaration(reactFunction.parent) &&
                  reactFunction.parent.name.getText());
              framework = 'react';
              kind = PropKind.Function;
              if (reactFunction.parameters.length) {
                const props = reactFunction.parameters[0];
                if (ts.isObjectBindingPattern(props.name)) {
                  initializer = props.name;
                }
                propsType = checker.getTypeAtLocation(props);
              }
            }
          }
        }
      }
    }
    name =
      typeof displayName === 'string'
        ? displayName
        : displayName && ts.isStringLiteral(displayName)
        ? displayName.text
        : undefined;
    if (defaultProps) {
      initializer = defaultProps;
    }

    return {
      type: propsType,
      name,
      initializer,
      kind,
      framework,
      skipGenerics: kind === PropKind.Function,
      skipParameters: kind === PropKind.Function,
    };
  },
};

export default reactPLugin;
