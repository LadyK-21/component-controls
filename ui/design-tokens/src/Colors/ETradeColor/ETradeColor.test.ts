import path from 'path';
import MatchMediaMock from 'jest-matchmedia-mock';
import { loadConfigurations } from '@component-controls/config';
import { renderExample } from '@component-controls/test-renderers';
import { render as reactRender } from '@component-controls/render/react';
import { render, act } from '@testing-library/react';

const renderErr = () => {
  throw new Error('Could not render the story');
};

import doc, { overview, palette } from './ETradeColor.stories';

describe('ETradeColor', () => {
  let matchMedia: MatchMediaMock;
  beforeAll(() => {
    jest.mock('rc-util/lib/Portal');
    matchMedia = new MatchMediaMock();
  });
  afterEach(() => {
    matchMedia.clear();
  });
  const configPath = path.resolve(__dirname, '../../../.config');
  const config = loadConfigurations(configPath);
  if (!config.renderFn) {
    config.renderFn = reactRender;
  }

  test('overview', () => {
    const example = overview;

    let rendered;
    act(() => {
      rendered = renderExample({
        example,
        doc,
        config,
      });
    });
    if (!rendered) {
      renderErr();
      return;
    }
    const { asFragment } = render(rendered);
    expect(asFragment()).toMatchSnapshot();
  });

  test('palette', () => {
    const example = palette;

    let rendered;
    act(() => {
      rendered = renderExample({
        example,
        doc,
        config,
      });
    });
    if (!rendered) {
      renderErr();
      return;
    }
    const { asFragment } = render(rendered);
    expect(asFragment()).toMatchSnapshot();
  });
});
