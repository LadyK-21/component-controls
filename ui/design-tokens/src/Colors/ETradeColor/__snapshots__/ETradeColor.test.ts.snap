// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ETradeColor overview 1`] = `
<DocumentFragment>
  <div
    class="css-dx0ghv-ETradeColor"
  >
    <div
      class="css-1ejv39n-ETradeColor"
    >
      <div
        class="css-1hwz39e-ETradeColor"
      >
        <div
          class="css-esr948-ETradeColor"
        >
          Negative red
        </div>
        <div>
          #cc0000
        </div>
        <div />
        <div>
          $negative
        </div>
      </div>
    </div>
    <div
      class="css-1hgxxih-ETradeColor"
    >
      <div
        class="css-m3719-ETradeColor"
      >
        <div>
          Error state
        </div>
        <div
          class="css-144hmk0-ETradeColor"
        >
          Data trending down
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ETradeColor palette 1`] = `
<DocumentFragment>
  <div
    class="css-1gc1jl3"
  >
    <div
      class="css-dx0ghv-ETradeColor"
    >
      <div
        class="css-1610yxw-ETradeColor"
      >
        <div
          class="css-1hwz39e-ETradeColor"
        >
          <div
            class="css-esr948-ETradeColor"
          >
            Purple
          </div>
          <div>
            #5627d8
          </div>
          <div />
          <div>
            $link-color
          </div>
        </div>
      </div>
      <div
        class="css-1hgxxih-ETradeColor"
      >
        <div
          class="css-m3719-ETradeColor"
        >
          <div />
          <div
            class="css-144hmk0-ETradeColor"
          >
            Text links and other interactive elements
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-dx0ghv-ETradeColor"
    >
      <div
        class="css-1ejv39n-ETradeColor"
      >
        <div
          class="css-1hwz39e-ETradeColor"
        >
          <div
            class="css-esr948-ETradeColor"
          >
            Negative red
          </div>
          <div>
            #cc0000
          </div>
          <div />
          <div>
            $negative
          </div>
        </div>
      </div>
      <div
        class="css-1hgxxih-ETradeColor"
      >
        <div
          class="css-m3719-ETradeColor"
        >
          <div>
            Error state
          </div>
          <div
            class="css-144hmk0-ETradeColor"
          >
            Data trending down
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-dx0ghv-ETradeColor"
    >
      <div
        class="css-18m5wxq-ETradeColor"
      >
        <div
          class="css-1hwz39e-ETradeColor"
        >
          <div
            class="css-esr948-ETradeColor"
          >
            Positive green
          </div>
          <div>
            #107a1b
          </div>
          <div>
            .text-positive
          </div>
          <div>
            $positive
          </div>
        </div>
      </div>
      <div
        class="css-1hgxxih-ETradeColor"
      >
        <div
          class="css-m3719-ETradeColor"
        >
          <div>
            Success state
          </div>
          <div
            class="css-144hmk0-ETradeColor"
          >
            Data trending up
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-dx0ghv-ETradeColor"
    >
      <div
        class="css-1xrvtjh-ETradeColor"
      >
        <div
          class="css-1hwz39e-ETradeColor"
        >
          <div
            class="css-esr948-ETradeColor"
          >
            Body gray
          </div>
          <div>
            #242424
          </div>
          <div>
            .body-text-color
          </div>
          <div>
            $body-text-color
          </div>
        </div>
      </div>
      <div
        class="css-1hgxxih-ETradeColor"
      >
        <div
          class="css-m3719-ETradeColor"
        >
          <div />
          <div
            class="css-144hmk0-ETradeColor"
          >
            Headers and body text
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-dx0ghv-ETradeColor"
    >
      <div
        class="css-mnab2t-ETradeColor"
      >
        <div
          class="css-1hwz39e-ETradeColor"
        >
          <div
            class="css-esr948-ETradeColor"
          >
            Background accent color
          </div>
          <div>
            #f1ecff
          </div>
          <div>
            .background-accent-color-1
          </div>
          <div>
            $background-accent-color-1
          </div>
        </div>
      </div>
      <div
        class="css-1hgxxih-ETradeColor"
      >
        <div
          class="css-m3719-ETradeColor"
        >
          <div />
          <div
            class="css-144hmk0-ETradeColor"
          >
            Background color
          </div>
          <div
            class="css-esr948-ETradeColor"
          >
            <svg
              aria-hidden="true"
              class="css-1xoidnz-ETradeColor"
              fill="currentColor"
              height="16"
              role="img"
              style="display: inline-block; user-select: none; vertical-align: text-bottom;"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M1.5 8a6.5 6.5 0 0110.535-5.096l-9.131 9.131A6.472 6.472 0 011.5 8zm2.465 5.096a6.5 6.5 0 009.131-9.131l-9.131 9.131zM8 0a8 8 0 100 16A8 8 0 008 0z"
                fill-rule="evenodd"
              />
            </svg>
            Not for text
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
